src/components/Tables/ButtonFilter.tsx

```diff
@@ -0,0 +1,75 @@
+"use client";
+
+import React from "react";
+import { ClientSideFilter, ClientSideFilterMethod, defaultToString } from "./Filters";
+import Button from "@mui/material/Button";
+import { capitaliseWords } from "@/common/format";
+
+interface ButtonGroupFilterProps<Data> {
+    key: keyof Data;
+    filterLabel: string;
+    filterOptions: string[];
+    initialActiveFilter: string;
+    method: ClientSideFilterMethod<Data, string>;
+    persistOnClear: boolean;
+}
+
+interface ButtonProps {
+    activeFilter: string;
+    buttonLabel: string;
+    setState: (state: string) => void;
+}
+
+const FilterButton: React.FC<ButtonProps> = (buttonProps) => {
+    const active = buttonProps.activeFilter === buttonProps.buttonLabel;
```

Comment:
very minor point, but I'm a fan of boolean naming conventions where boolean names start with either `is` or `should` (or similar). So here I'd prefer `isActive` over `active`